beetsplug/mbsubmit.py

```diff
@@ -18,31 +18,334 @@
 parseable by the MusicBrainz track parser [1]. Programmatic submitting is not
 implemented by MusicBrainz yet.
 
+The plugin also allows the user to open the tracks in MusicBrainz Picard [2].
+
+Another option this plugin provides is to help with creating a new release
+on MusicBrainz by seeding the MusicBrainz release editor [3]. This works in
+the following way:
+
+- Host a small web server that serves a web page. When loaded by the user,
+  this page will automatically POST data to MusicBrainz as described in [3].
+- The same web server also listens for a callback from MusicBrainz, see
+  redirect_uri [3] and will try to import an album using the newly created
+  release.
+- jwt tokens with random keys are used to prevent using this web server in
+  unintended ways.
+
+This feature is loosely based on how browser integration is implemented in
+Picard [4].
+
 [1] https://wiki.musicbrainz.org/History:How_To_Parse_Track_Listings
+[2] https://picard.musicbrainz.org/
+[3] https://musicbrainz.org/doc/Development/Seeding/Release_Editor
+[4] https://github.com/metabrainz/picard/blob/master/picard/browser/browser.py
 """
-
 import subprocess
+import threading
+import time
+import uuid
+import webbrowser
+from collections import defaultdict
+from dataclasses import dataclass
+from html import escape
+from http.server import BaseHTTPRequestHandler, ThreadingHTTPServer
+from secrets import token_bytes
+from typing import Optional
+from urllib.parse import parse_qs, urlparse
 
-from beets import ui
+from beets import autotag, ui
 from beets.autotag import Recommendation
 from beets.plugins import BeetsPlugin
+from beets.ui import print_
 from beets.ui.commands import PromptChoice
 from beets.util import displayable_path
+from beets.util.pipeline import PipelineThread
 from beetsplug.info import print_data
 
+try:
+    import jwt
+except ImportError:
+    jwt = None
+
+_form_template = """<!doctype html>
+<meta charset="UTF-8">
+<html>
+<body>
+    <form action="{action}" method="post" style="display: none">
+        {form_data}
+        <input type="submit">
+    </form>
+    <script>document.forms[0].submit()</script>
+</body>
+"""
+
+_form_input_template = '<input type="hidden" name="{name}" value="{value}">'
+
+
+@dataclass
+class CreateReleaseTask:
+    """
+    Represents a task for creating a single release on MusicBrainz and its current
+    status.
+    """
+
+    formdata: dict
+    """
+    Form data to be submitted to MB.
+    """
+
+    result_release_mbid: Optional[str] = None
+    """
+    Contains the release ID returned by MusicBrainz after the release was created.
+    """
+
+    browser_opened: bool = False
+    """
+    True when the user has opened the link for this task in the browser.
+    """
```

Comment:
Docstring comments go above variables. It's only classes and functions that have them below the definition. Confusing, I know

Reply:
Do they? See https://peps.python.org/pep-0258/#attribute-docstrings... also, PyCharm correctly picks up the docstrings this way.